### PR TITLE
Fp16 option

### PR DESCRIPTION
**Pull request by okdshin**
This PR adds `allow_fp16_mode` and `force_fp16_mode` options
`allow_fp16_mode` doesn't always use fp16 kernel.
`force_fp16_mode` always tries to use fp16 kernel and when device doesn't support fp16 mode, it throws error.
This PR depeneds on #201

### CODE REVIEW

File: menoh/tensorrt/Inference.cpp [hunk 1]
```diff
@@ -219,37 +214,52 @@ namespace menoh_impl {
...
-            builder->setFp16Mode(false);
+            if(config_.force_fp16_mode) {
+                if(!builder->platformHasFastFp16()) {
+                    throw ParseException(
+                      "FP16 mode is not available on this device");
+                }
+                builder->setFp16Mode(true);
+                builder->setStrictTypeConstraints(true);
+            }
+            if(config_.allow_fp16_mode && builder->platformHasFastFp16()) {
```
Comment (msakai) [inline menoh/tensorrt/Inference.cpp]: `else if` ?

File: menoh/tensorrt/Inference.cpp [hunk 2]
```diff
@@ -321,19 +324,28 @@ namespace menoh_impl {
...
+                         .count() /
+                       1000.0
+                  << " sec" << std::endl;
+
+                gProfiler.printLayerTimes();
+                std::cout << "Inference::Run::done" << std::endl;
+            } else {
+                runner();
+            }
 #endif
```
Comment (msakai) [inline menoh/tensorrt/Inference.cpp]: If `MENOH_ENABLE_TENSORRT_PROFILER` is not defined `runner` will not be executed.
Is that intended behavior?

File: menoh/tensorrt/model_core.cpp [hunk 3]
```diff
@@ -41,10 +42,22 @@ namespace menoh_impl {
...
                     }
+                    if(c.find("enable_profiler") != c.end()) {
+                        enable_profiler = c["enable_profiler"].get<int>();
+                    }
+                    if(c.find("allow_fp16_mode") != c.end()) {
+                        allow_fp16_mode = c["allow_fp16_mode"].get<int>();
+                    }
+                    if(c.find("force_fp16_mode") != c.end()) {
+                        force_fp16_mode = c["force_fp16_mode"].get<int>();
+                    }
```
Comment (msakai) [inline menoh/tensorrt/model_core.cpp]: JSON has boolean values.
If `get<bool>` method exists, then I feel it is more appropriate than `get<int>`.

File: menoh/tensorrt/Inference.cpp (hunk 2 above)
Comment (okdshin) [reply menoh/tensorrt/Inference.cpp]: No. I fixed it #201. I'll merge it to this PR.

File: menoh/tensorrt/model_core.cpp (hunk 3 above)
Comment (okdshin) [reply menoh/tensorrt/model_core.cpp]: Oh I didn't know it. I'll fix.

**Comment by okdshin** [comment]
I fixed. Please check @msakai

**Comment by okdshin** [comment]
Thank you fore reviewing. I checked this on my environment with gpu. I'll merge.

**Pull request merged**